// nutrition-ai/dist/index.mjs
// src/generate.ts
import { GoogleGenerativeAI } from "@google/generative-ai";
import { z } from "zod";

// src/template.ts
var prompt = `As an expert nutritionist, analyze the food dish presented in the image, applying your specialized knowledge to identify each food item and its caloric content. Use Markdown formatting to structure your response clearly and attractively. Your analysis should encompass the following components:

### Food Item Analysis

**1. [Item Name] : [Number] calories ** 

Provide a short overview of its nutritional value and health benefits, using your expertise as a nutritionist.

### Total Calories

Calculate the sum of calories for all identified items, showcasing your ability to evaluate the meal's energy content comprehensively.

### Nutritional Composition Breakdown

Break down the nutritional composition, emphasizing the percentages for macronutrients and their roles in a healthy diet.
**Carbohydrates:** [Percentage]%
**Fats:** [Percentage]%
**Proteins:** [Percentage]%
**Fibers:** [Percentage]%
**Sugars:** [Percentage]%

### Overall Nutritional Value

Assess the meal's nutritional balance and healthfulness from your professional standpoint. Offer personalized dietary advice, including healthier alternatives and modifications that reflect your expert knowledge.

### Dietary Recommendations

Offer actionable advice for enjoying the meal in a healthy manner, leveraging your expertise to suggest portion sizes, healthier ingredient alternatives, and strategies for balancing this meal within a day's diet.

If you are unable to identify the contents of the uploaded image or if it is not related to food items, please state:

'I cannot identify the contents of the uploaded image. Please upload a clear and relevant image of food items.'

### Conclusion
Conclude with general advice on maintaining a healthy diet, emphasizing the importance of variety, moderation, and nutrient density. Your conclusion should reflect the comprehensive and personalized care an expert nutritionist would offer.

Your response, formatted with Markdown for clarity and engagement, should mirror the depth of knowledge and personalized attention expected from a nutritionist consultation.`;

// src/generate.ts
async function generateAnswer(input) {
  const validated = validateInputSchema.safeParse(input);
  if (!validated.success) {
    throw new Error(
      `Invalid parameters: ${JSON.stringify(
        validated.error.flatten().fieldErrors
      )}`
    );
  }
  const {
    fileBase64,
    fileMimeType,
    googleKey,
    model = "gemini-pro-vision"
  } = validated.data;
  const imagePart = {
    inlineData: {
      data: fileBase64,
      mimeType: fileMimeType
    }
  };
  try {
    const genAI = new GoogleGenerativeAI(googleKey);
    const genModel = genAI.getGenerativeModel({ model });
    const result = await genModel.generateContent([prompt, imagePart]);
    const response = await result.response;
    return response.text();
  } catch (error) {
    throw error;
  }
}
var validateInputSchema = z.object({
  googleKey: z.string({
    required_error: "googleKey is required."
  }).min(1, {
    message: "Invalid googleKey."
  }),
  model: z.string().optional(),
  fileBase64: z.string({
    required_error: "fileBase64 is required."
  }).min(1, {
    message: "Invalid fileBase64."
  }),
  fileMimeType: z.enum(["image/jpeg", "image/png"], {
    required_error: "fileMimeType is required."
  })
});
export {
  generateAnswer
};
